/// \page exampleR
# command to load the functions for use in R console:  (where directory1/directory2/ is the absolute path)
# source("/directory1/directory2/Visualization_methods_Rcode.R")
#
#
# The command  view.help()  used in the R console will provide a description of the input to the various arguments, and any output (other than the plot output).
# The remaining 4 functions are methods for visualizing transcription factor binding site data, initially for ChIP-Seq but not limited to this data type.
# The CB.plot, landscape.view, and bimotif.view require, minimally, that you provide both a data file and the columns containing the pertinent data, while the dinculeotide.view only requires, minimally, an alignment file (generated by provided perl code) as input.
# 
# view.help()
#


# display functions and options to user in R console
cat(sep="\n")
cat( "+ Help on details of functions:", "view.help()", sep="\n")
cat(sep="\n")
cat( "+ Composition-bias plot:", paste("CB.plot( data.file=NA, col.X=0, col.Y=0, col.Y.pval=F, yaxis.lim=F, col.TFnames=0, headerline=T, title.plot=\"\", save.plot.to=F  )", sep="\t"),  sep="\n" )
cat(sep="\n")
cat( "+ TFBS-landscape view:", paste("landscape.view( data.file=NA, col.X=0, col.Y=0, score.type=\"relative\", yaxis.lim1=F, yaxis.lim2=F, headerline=T,  title.plot=\"\", save.plot.to=F, threshold=F, hadb.plot=F , hadb.resid.fold=2, hadb.score.diff=0.20 )", sep="\t"), sep="\n" )
cat(sep="\n")
cat( "+ TFBS-bi-motif view:", paste( "bimotif.view(  data.file=NA, col.X=0, col.Y=0, resolution=5, yaxis.lim=F, headerline=T, title.plot=\"\", save.plot.to=F  )", sep="\t"), sep="\n" )
cat(sep="\n")
cat( "+ Dinucleotide-envinronment view:", paste( "dinucleotide.view( data.file=NA,  x1.adj=100, x2.adj=100, yaxis.lim=F, title.plot=\"\", save.plot.to=F  )", sep="\t"),  sep="\n" )
cat(sep="\n")

	
#-----------------------------------
# Help Function
#-----------------------------------
view.help = function(){
	options(width=120)
	cat("Usage: ", sep="\n")
	cat( "Note: ") ;
	writeLines(strwrap("1) non-logical and non-numeric values should be in double quotes e.g. title.plot=\"My plot 1\"",  indent=0, exdent=0, initial="\t", prefix="\t\t   ") )
	writeLines(strwrap("2) logical values are T for true and F for false", indent=0, exdent=0, initial="\t\t", prefix="\t\t   ") )
	writeLines(strwrap("3) if identifying the columns by name rather than by number, column names in the data file must consist of either alphanumeric or '_', because R converts some characters, like spaces or hyphens, to dots '.' when they are present in a column name. A '#' at the start of a column name is converted to 'X.'", indent=0, exdent=0, initial="\t\t", prefix="\t\t   ") )

